### PR TITLE
fix(manifests): Change mysql default binlog expire to 10 days. Fixes #8552

### LINKED ISSUE #8552: MySQL logs disk full error
We noticed on some KFP cluster listing run failed consistently with 524 timeout error.

Looking at the pod logs, metadata-writer and persistence-agent are in crash loop with error failed to connect to mysql.
Checking mysql pod logs, it's flooded with
> `disk is full writing './binlog.~rec~'`

Did some research, looks like MySQL 8.0 changed the default log expiration from 10 days to 30 days: 
https://dba.stackexchange.com/questions/233048/my-disk-is-full-of-binlog-files

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
Fixes #8552 

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: manifests/kustomize/third-party/mysql/base/mysql-deployment.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ spec:
         # https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_default_authentication_plugin
         # https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_authentication_policy
         - --default-authentication-plugin=mysql_native_password
+        - --binlog-expire-logs-seconds=864000
```
Comment (gkcalat) [inline manifests/kustomize/third-party/mysql/base/mysql-deployment.yaml:45]: NIT: should we add a comment here describing the settings and pointing to the docs: https://dev.mysql.com/doc/refman/8.0/en/replication-options-binary-log.html#sysvar_binlog_expire_logs_seconds ?

**Review by gkcalat — COMMENTED** [review]
Thank you @chensun! Should we add this to `release-1.8` branch?

File: manifests/kustomize/third-party/mysql/base/mysql-deployment.yaml (hunk 1 above)
Comment (chensun) [reply manifests/kustomize/third-party/mysql/base/mysql-deployment.yaml:45]: Good suggestion, I'll add the comment.

**Review by gkcalat — APPROVED** [review]

**Comment by chensun** [comment]
> Thank you @chensun! Should we add this to `release-1.8` branch?

It looks like we didn't upgrade mysql version in `release-1.8`.

**Comment by gkcalat** [comment]
> > Thank you @chensun! Should we add this to `release-1.8` branch?
> 
> It looks like we didn't upgrade mysql version in `release-1.8`.

Oh yes, I recall now :)

**Pull request merged**